src/Modules/OneTapLogin.php

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * One Tap Login Class.
+ *
+ * This class will be responsible for handling
+ * Google's one tap login for web functioning.
+ *
+ * @package RtCamp\GoogleLogin\Modules
+ * @since 1.0.16
+ */
+
+declare(strict_types=1);
+
+namespace RtCamp\GoogleLogin\Modules;
+
+use Exception;
+use RtCamp\GoogleLogin\Utils\Authenticator;
+use RtCamp\GoogleLogin\Utils\GoogleClient;
+use RtCamp\GoogleLogin\Utils\Helper;
+use RtCamp\GoogleLogin\Interfaces\Module;
+use RtCamp\GoogleLogin\Utils\TokenVerifier;
+use function RtCamp\GoogleLogin\plugin;
+
+/**
+ * Class OneTapLogin
+ *
+ * @package RtCamp\GoogleLogin\Modules
+ */
+class OneTapLogin implements Module {
+	/**
+	 * Settings Module.
+	 *
+	 * @var Settings
+	 */
+	private $settings;
+
+	/**
+	 * Token verifier.
+	 *
+	 * @var TokenVerifier
+	 */
+	private $token_verifier;
+
+	/**
+	 * Google client instance.
+	 *
+	 * @var GoogleClient
+	 */
+	private $google_client;
+
+	/**
+	 * Authenticator service.
+	 *
+	 * @var Authenticator
+	 */
+	private $authenticator;
+
+	/**
+	 * OneTapLogin constructor.
+	 *
+	 * @param Settings      $settings Settings object.
+	 * @param TokenVerifier $verifier Token verifier object.
+	 * @param GoogleClient  $client   Google client instance.
+	 * @param Authenticator $authenticator Authenticator service instance.
+	 */
+	public function __construct( Settings $settings, TokenVerifier $verifier, GoogleClient $client, Authenticator $authenticator ) {
+		$this->settings       = $settings;
+		$this->token_verifier = $verifier;
+		$this->google_client  = $client;
+		$this->authenticator  = $authenticator;
+	}
+
+	/**
+	 * Module name.
+	 *
+	 * @return string
+	 */
+	public function name(): string {
+		return 'one_tap_login';
+	}
+
+	/**
+	 * Module Initialization activity.
+	 *
+	 * Everything will happen if and only if one tap is active in settings.
+	 */
+	public function init(): void {
+		if ( $this->settings->one_tap_login ) {
+			add_action( 'login_enqueue_scripts', [ $this, 'one_tap_scripts' ] );
+			add_action( 'login_footer', [ $this, 'one_tap_prompt' ] );
+			add_action( 'wp_ajax_nopriv_validate_id_token', [ $this, 'validate_token' ] );
+			add_action( 'rtcamp.id_token_verified', [ $this, 'authenticate' ] );
+			add_action(
+				'init',
+				function () {
+					if ( ! is_user_logged_in() ) {
+						add_action( 'wp_enqueue_scripts', [ $this, 'one_tap_scripts' ] );
+						add_action( 'wp_footer', [ $this, 'one_tap_prompt' ], 10000 );
+					}
+				} 
+			);
+		}
+	}
+
+	/**
+	 * Show one tap prompt markup.
+	 *
+	 * @return void
+	 */
+	public function one_tap_prompt(): void { ?>
+		<div id="g_id_onload"
+			 data-client_id="<?php echo esc_html( $this->settings->client_id ); ?>"
+			 data-login_uri="<?php echo esc_html( wp_login_url() ); ?>"
+			 data-callback="LoginWithGoogleDataCallBack"
+		</div>
```

Comment:
Suggestion: You can use `printf()` to echo HTML instead of closing/opening PHP tag for small HTML.

Reply:
Noted, but keeping HTML separate from PHP seems better idea to me. 

If we ever need to move markup to some different templates, this practice would be very handy.